subgraphs/burrow/src/handlers/liquidate.ts

```diff
@@ -0,0 +1,402 @@
+            /* -------------------------------------------------------------------------- */
+            const in_assets = liqCall.toObject().get("in_assets");
+            if (in_assets && in_assets.kind == JSONValueKind.ARRAY) {
+              if (in_assets.toArray()[0].kind == JSONValueKind.OBJECT) {
+                const asset = in_assets.toArray()[0].toObject();
+                const asset_id = asset.get("token_id");
+                const asset_amt = asset.get("amount");
+                if (asset_id && asset_amt) {
+                  token_in = asset_id.toString();
+                  token_in_amount = asset_amt.toString();
+                }
+              }
+            }
+            /* -------------------------------------------------------------------------- */
+            /*                            Collateral asset: id & amount                   */
+            /* -------------------------------------------------------------------------- */
+            const out_assets = liqCall.toObject().get("out_assets");
+            if (out_assets && out_assets.kind == JSONValueKind.ARRAY) {
+              if (out_assets.toArray()[0].kind == JSONValueKind.OBJECT) {
+                const asset = out_assets.toArray()[0].toObject();
+                const asset_id = asset.get("token_id");
+                const asset_amt = asset.get("amount");
+                if (asset_id && asset_amt) {
+                  token_out = asset_id.toString();
+                  token_out_amount = asset_amt.toString();
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  if (token_in && token_out && token_in_amount && token_out_amount) {
+    const repaidMarket = getOrCreateMarket(token_in);
+    const repaidPosition = getOrCreatePosition(
+      liquidatee,
+      repaidMarket,
+      "BORROWER",
+      receipt
+    );
+    const dailySnapshot = getOrCreateMarketDailySnapshot(repaidMarket, receipt);
+    const hourlySnapshot = getOrCreateMarketHourlySnapshot(
+      repaidMarket,
+      receipt
+    );
+    const collateralMarket = getOrCreateMarket(token_out);
+    const collateralPosition = getOrCreatePosition(
+      liquidatee,
+      collateralMarket,
+      "LENDER",
+      receipt
+    );
+
+    repaidPosition.balance = BI_ZERO;
+    repaidPosition.hashClosed = receipt.receipt.id.toBase58();
+    repaidPosition.blockNumberClosed = BigInt.fromU64(
+      receipt.block.header.height
+    );
+    repaidPosition.timestampClosed = BigInt.fromU64(
+      receipt.block.header.timestampNanosec / 1000000000
+    );
+
+    collateralPosition.balance = BI_ZERO;
+    collateralPosition.hashClosed = receipt.receipt.id.toBase58();
+    collateralPosition.blockNumberClosed = BigInt.fromU64(
+      receipt.block.header.height
+    );
+    collateralPosition.timestampClosed = BigInt.fromU64(
+      receipt.block.header.timestampNanosec / 1000000000
+    );
+
+    liquidatee.openPositionCount -= 1;
```

Comment:
how do you know the full position was liquidated?

Reply:
bc there's no partial liquidation in burrow

Reply:
https://docs.burrow.cash/product-docs/how-burrow-works/liquidations#liquidation-example

Reply:
fixed